### PR TITLE
Add another lighting service (Razer)

### PR DESCRIPTION
**Pull request by dhobi**
Added Razer light service by:
- Creating another project "OnlineStatusLight.Application.Razer.RazerLightService"
- Refactoring to ILightService (RazerLightService and SonoffBasicR3Service implement this)

Place the switch in appsettings.json
```
// for Razer
"lightservice": "OnlineStatusLight.Application.Razer.RazerLightService, OnlineStatusLight.Application.Razer"

// for Sonoff (default)
"lightservice": "OnlineStatusLight.Application.SonoffBasicR3Service, OnlineStatusLight.Application"
```

Razer has a config to only light up the headset instead of all devices
```
"razer": {
    // true = will color the HeadSet device only. false = will color ALL Razer devices
    "headsetonly": true
}
```

### CODE REVIEW

File: OnlineStatusLight.Forms/appsettings.json [hunk 1]
```diff
@@ -9,6 +13,6 @@
   },
   "msteams": {
     "interval": 5,
-    "logfile": "C:\\Users\\miscalge\\AppData\\Roaming\\Microsoft\\Teams\\logs.txt"
+    "logfile": "%appdata%\\Microsoft\\Teams\\logs.txt"
```
Comment (miscalencu) [inline OnlineStatusLight.Forms/appsettings.json:16]: nice!

File: OnlineStatusLight.Forms/Startup.cs [hunk 2]
```diff
@@ -60,6 +61,11 @@ private static void ProcessExit(object? sender, EventArgs e)
 
         public static IServiceCollection ConfigureHttpClients(IServiceCollection services, IConfiguration configuration)
         {
+            if(configuration["sonoff"] == null)
```
Comment (miscalencu) [inline OnlineStatusLight.Forms/Startup.cs:64]: this check here is failing.
maybe you can remove this bit and instead add a check before calling this method, like:

```
                    // add sonoff http clients
                    if (lightService == typeof(SonoffBasicR3Service))
                        services = ConfigureHttpClients(services, ConfigurationRoot);
```

and even rename the ConfigureHttpClients to ConfigureSonoffHttpClients or something similar.

**Comment by miscalencu** [comment]
Nice addition, thanks! Please check the comment I added and once that is fixed I will approve the PR.

**Comment by dhobi** [comment]
I only see one (resolved) comment from you on the %appdata% change. I don't think you were referring to this, were you?

**Comment by miscalencu** [comment]
> I only see one (resolved) comment from you on the %appdata% change. I don't think you were referring to this, were you?

Sorry, just added it now.

File: OnlineStatusLight.Forms/Startup.cs (hunk 2 above)
Comment (dhobi) [reply OnlineStatusLight.Forms/Startup.cs:64]: Hmm yeah, I see what you mean. Ideally, the configuration of the sonoff httpclients would happen in the application specific project anyway as the razer implementation does not need it at all. Like this, it seems the parent knows the specifics of its children...

**Pull request merged**